### PR TITLE
update for 4572-update multiple fingerprints

### LINKED ISSUE #305: Update to draft-ietf-mmusic-4572-update-05
This changes the requirements around use of `a=fingerprint`.

### LINKED ISSUE #306: Add reference to draft-ietf-mmusic-4572-update
This updates 4572 so needs to be referenced. 

Also see 
https://github.com/rtcweb-wg/jsep/issues/305

### PR DESCRIPTION
**Pull request by fluffy**
Fixes #306 and Fixes #305

### CODE REVIEW

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -2090,10 +2090,11 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
                     <t>For DTLS, all m= sections MUST use the
-                    certificate for the identity that has been specified
-                    for the PeerConnection; as a result, they MUST all
-                    have the same <xref target="RFC4572"></xref>
-                    fingerprint value, or this value MUST be a
+                    certificate for the identity that has been
+                    specified for the PeerConnection; as a result,
+                    they MUST all have the same <xref
+                    target="I-D.ietf-mmusic-4572-update"></xref>
+                    fingerprint value(s), or this value MUST be a
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: this value MUST be a session-level attribute -> these value(s) MUST be session-level attributes

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -2196,11 +2197,14 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                         specified in <xref target="RFC5245"></xref>,
                         Section 15.4.</t>
 
-                        <t>An "a=fingerprint" line for each of the
-                        endpoint's certificates, as specified in <xref
-                        target="RFC4572"></xref>, Section 5; the digest
-                        algorithm used for the fingerprint MUST match
-                        that used in the certificate signature.</t>
+                        <t>One or more "a=fingerprint" line(s) for
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: q: is multiple certificates still something we support?

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -2196,11 +2197,14 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+                        each of the endpoint's certificates, as
+                        specified in <xref
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I think you should include some text about the hash function to make the rationale for multi fingerprints more clear, e.g.

For each of the endpoint's certificates, and for each supported digest algorithm, an "a=fingerprint" line computed for the specific certificate and digest algorithm. At least...

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -2893,11 +2897,14 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                         specified in <xref target="RFC5245"></xref>,
                         Section 15.4.</t>
 
-                        <t>An "a=fingerprint" line for each of the
-                        endpoint's certificates, as specified in <xref
-                        target="RFC4572"></xref>, Section 5; the digest
-                        algorithm used for the fingerprint MUST match
-                        that used in the certificate signature.</t>
+                        <t>One or more "a=fingerprint" line(s) for
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: see above

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: yes - so we can support different public key types - for example  RSA vs EC

File: draft-ietf-rtcweb-jsep.xml (hunk 1 above)
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: fixed

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -2196,11 +2197,14 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+                        each of the endpoint's certificates, as
+                        specified in <xref
+                        target="I-D.ietf-mmusic-4572-update"></xref>.
+                        At least one of the "a=fingerprint" lines for
+                        each certificate MUST have use the digest
+                        algorithm used in the certificate
+                        signature.</t>
```
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: It's not in 4572 update yet, but I think this is all going to end up there. What do you think of just remove the part with "At least one of the "a=fingerprint" lines for
 +                        each certificate MUST have use the digest
 +                        algorithm used in the certificate
 +                        signature" and leaging that to 4572-update.

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: right

File: draft-ietf-rtcweb-jsep.xml (hunk 1 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: needs update for multi certificates

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -2089,11 +2089,12 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                     contain any ICE credentials and MUST NOT gather any
                     candidates.</t>
 
-                    <t>For DTLS, all m= sections MUST use the
-                    certificate for the identity that has been specified
-                    for the PeerConnection; as a result, they MUST all
-                    have the same <xref target="RFC4572"></xref>
-                    fingerprint value, or this value MUST be a
+                    <t>For DTLS, all m= sections MUST use all the
+                    certificate(s) for the identity that has been
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: suggest just  "use all the certificate(s) that have been specified for the PeerConnection"

File: draft-ietf-rtcweb-jsep.xml [hunk 7]
```diff
@@ -2089,11 +2089,12 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+                    specified for the PeerConnection; as a result,
+                    they MUST all have the same <xref
+                    target="I-D.ietf-mmusic-4572-update"></xref>
+                    fingerprint value(s), or these value(s) MUST be a
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: MUST be session-level attributes

File: draft-ietf-rtcweb-jsep.xml (hunk 6 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: adding to review

Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: suggest just "use all the certificate(s) that have been specified for the PeerConnection

File: draft-ietf-rtcweb-jsep.xml (hunk 7 above)
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: fixed

File: draft-ietf-rtcweb-jsep.xml (hunk 6 above)
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: fixed

**Comment by juberti** [comment]
whoops, didn't realize that eric hadn't reviewed before I merged.

@ekr, want to take a quick look?

**Pull request merged**